src/main/java/com/google/cloud/spanner/pgadapter/parsers/BooleanParser.java

```diff
@@ -34,6 +36,15 @@
 public class BooleanParser extends Parser<Boolean> {
   private static final String TRUE_VALUE = "t";
   private static final String FALSE_VALUE = "f";
+  private static final byte[] TRUE_VALUE_BYTES = new byte[] {'t'};
+  private static final byte[] FALSE_VALUE_BYTES = new byte[] {'f'};
+  private static final byte[] TRUE_VALUE_BYTES_BINARY = new byte[1];
+  private static final byte[] FALSE_VALUE_BYTES_BINARY = new byte[1];
+
+  static {
+    ByteConverter.bool(TRUE_VALUE_BYTES_BINARY, 0, true);
+    ByteConverter.bool(FALSE_VALUE_BYTES_BINARY, 0, true);
```

Comment:
Why are we passing `true` as the third parameter here? Shouldn't it be `false`?

Reply:
Great catch, thanks! That's absolutely right, and it was not caught by our tests for two reasons:
1. Most drivers will never use the binary format for booleans, because it does not really save any network traffic anyways (boolean values in text format are represented by `t` and `f`, so they take up just one byte, just as the binary format of 1 and 0)
2. Our tests mostly used `true` as the test value, which meant that they did not catch the fact that `false` was encoded as true.

I've added additional tests that use randomly generated values so we can catch this kind of errors in the future. That testing also turned out to detect a bug in the PostgreSQL JDBC driver, as that driver does not correctly decode binary boolean values. I've filed an issue and a PR for the JDBC driver in their repository for it.